libs/community/langchain_community/vectorstores/falkordb_vector.py

```diff
@@ -0,0 +1,1684 @@
+            ) -> None:
+        """
+        This method creates a new vector index
+        on a node in FalkorDB.
+        """
+        if node_label:
+            node_label = node_label
+        elif self.node_label:
+            node_label = self.node_label
+        else:
+            raise ValueError(
+                "`node_label` property must be set to use this function"
+            )
+        
+        if embedding_node_property:
+            embedding_node_property = embedding_node_property
+        elif self.embedding_node_property:
+            embedding_node_property = self.embedding_node_property
+        else:
+            raise ValueError(
+                "`embedding_node_property` property must be set to use this function"
+            )
+        
+        if embedding_dimension:
+            embedding_dimension = embedding_dimension
+        elif self.embedding_dimension:
+            embedding_dimension = self.embedding_dimension
+        else:
+            raise ValueError(
+                "`node_label` property must be set to use this function"
+            )
+        try:
+            self._database.create_node_vector_index(
+                node_label,
+                embedding_node_property,
+                dim=embedding_dimension,
+                similarity_function=DISTANCE_MAPPING[self._distance_strategy]
+            )
+            self._index_type = 'NODE'
+        except Exception as e:
+            if "already indexed" in str(e):
+                print(
+                    f"A vector index on (:{node_label}"
+                    "{"
+                    f"{embedding_node_property}"
+                    "}"
+                    ")"
+                    " has already been created"
+                      )
+
+    def create_new_index_on_relationship(
+            self,
+            relation_type: str = "",
+            embedding_node_property: str = "" ,
+            embedding_dimension: int = None,
+    ) -> None:
+        """
+        This method creates an new vector index
+        on a relationship/edge in FalkorDB.
+        """
+        if not relation_type:
+            raise ValueError(
+                "`relation_type` must be set to use this function"
+            )
+        if not embedding_node_property:
+            raise ValueError(
+                "`embedding_node_property` must be set to use this function"
+            )
+        if not embedding_dimension:
+            raise ValueError(
+                "`embedding_dimension` must be set to use this function"
+            )
+        
+        try:
+            self._database.create_edge_vector_index(
+                relation_type,
+                embedding_node_property,
+                dim=embedding_dimension,
+                similarity_function=DISTANCE_MAPPING[DEFAULT_DISTANCE_STRATEGY]
+            )
+            self._index_type = 'RELATIONSHIP'
+        except Exception as e:
+            if "already indexed" in str(e):
```

Comment:
ignoring the exception here can lead to a problematic when debugging errors also using the print function maybe use logger?

Reply:
Alright, I've added the following code;
```
else:
    print(f"Error occured: {e}")
```